src/merkle/tiered_smt/bits.rs

```diff
@@ -0,0 +1,66 @@
+use super::{Felt, RpoDigest, TieredSmt, TreeIndex};
+
+pub struct BitsIterator<'a> {
+    elements: &'a [Felt],
+    idx: usize,
+    bit: u64,
```

Comment:
Agreed, fixed! I wonder if the CPU will allow optimizations for this case as it will probably be a register (that is either u32 or u64), but then we should leave the possibility open